### ISSUE TITLE
Versions crashes the app

### ISSUE THREAD

**Issue by liyuqian** [flutter/flutter#27610]
When I add 
[`print(versions.flutterEngineVersion);`](https://master-docs.flutter.io/flutter/dart-ui/Versions-class.html)  (or `versions.skiaVersion`) before `runApp` in flutter_gallery app, the app will crash on `flutter run`. Here's the error message

```
E/DartVM  ( 4367): ../../third_party/dart/runtime/vm/native_entry.cc: 214: error: Failed to resolve native function 'Versions_getVersions' in 'Versions._getVersions@177065589'
E/DartVM  ( 4367): 
E/DartVM  ( 4367): thread=4387, isolate=main.dart:main()(0x92f61a00)
E/DartVM  ( 4367):   [0x9578d127] Unknown symbol
E/DartVM  ( 4367):   [0x9578d127] Unknown symbol
E/DartVM  ( 4367):   [0x957f1885] Unknown symbol
E/DartVM  ( 4367): -- End of DumpStackTrace

Oops; flutter has exited unexpectedly.
Sending crash report to Google.
```

**Comment by liyuqian** [comment]
CC @iskakaushik @chinmaygarde

**Comment by chinmaygarde** [comment]
Lets test in profile as well as debug modes. Also, we should investigate why the unit test is different from the running application.

**Comment by dnfield** [comment]
I fixed this in flutter/engine#7726

**Comment by dnfield** [comment]
Closing this but feel free to open if there was something you think that's missing here.  The tests that should have covered this were acting funny because they said they were dart:ui instead of importing dart:ui.  It looks like that was copied from the window_hooks_integration test, which is probably where this was going to live at one point.  I opened https://github.com/flutter/flutter/issues/27628 to track refactoring that one.
